### PR TITLE
Add a contexts/README.md to explain the DID context versioning scheme.

### PR DESCRIPTION
**Pull request by dmitrizagidulin**
Add an explanation and changelog for the DID `@context` versioning scheme.

(Related to PR #270 and others.)

### CODE REVIEW

File: contexts/README.md [hunk 1]
```diff
@@ -0,0 +1,43 @@
+# DID Contexts
+
+## Background
+
+JSON-LD Context versioning (and schema versioning in general) is a complex 
+topic, involving the need to balance security and developer convenience.
```
Comment (TallTed) [inline contexts/README.md]: `security and developer`
-> `security with developer`

File: contexts/README.md [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+On one end of the spectrum, some use cases require cryptographically bound and 
+immutable contexts, and will need to use a contest-addressable URL mechanism,
```
Comment (TallTed) [inline contexts/README.md]: `contest-addressable`
-> `context-addressable`

Comment (dlongley) [inline contexts/README.md]:
```suggestion
immutable contexts, and will need to use a content-addressable URL mechanism,
```

Comment (dlongley) [reply contexts/README.md]: Note: You don't necessarily "need" to use a content-addressable URL mechanism... your software could instead force the resolution of a given URL to some hard-coded content (which is what will be expected for those using the v1 DID context once its done).

File: contexts/README.md (hunk 1 above)
Comment (dmitrizagidulin) [reply contexts/README.md]: Thanks!

File: contexts/README.md (hunk 2 above)
Comment (dmitrizagidulin) [reply contexts/README.md]: Good point, will add a note to that effect.

File: contexts/README.md [hunk 3]
```diff
@@ -6,9 +6,10 @@ JSON-LD Context versioning (and schema versioning in general) is a complex
 topic, involving the need to balance security with developer convenience.
 
 On one end of the spectrum, some use cases require cryptographically bound and 
-immutable contexts, and will need to use a content-addressable URL mechanism,
-such as the in-progress 
-[Hashlink](https://tools.ietf.org/html/draft-sporny-hashlink-03) spec.
+immutable contexts, and will need to enforce immutability at the code level
+(by using a content-addressable URL mechanism such as the in-progress 
+[Hashlink](https://tools.ietf.org/html/draft-sporny-hashlink-03) spec, or by
+forcing the resolution of a given URL to a known hard-coded embedded context).
```
Comment (TallTed) [inline contexts/README.md]: I think this should allow for as-yet-unknown code-level options, so `for example` should be added to start or end of the parenthetical.

Comment (dmitrizagidulin) [reply contexts/README.md]: Thanks, will update.

**Comment by OR13** [comment]
As I noted on https://github.com/w3c-ccg/did-spec/pull/270

Changes to the context, may cause broken links such as https://web-payments.org/vocabs/security#publicKeyJwk

It would be awesome to add a sentence or two about whether that matters or not, if it does matter, contributors should be guided to submit corrections for:

"sec": "https://w3id.org/security#",

They can read more about how to accomplish that here: https://github.com/perma-id/w3id.org

**Comment by brentzundel** [comment]
This repo is scheduled to be archived. The work has moved to [the DID WG](https://github.com/w3c/did-spec/pulls).
The artifacts in the DID WG repository share commit history with this repo, so it should be possible to raise this PR [against that repo](https://github.com/w3c/did-spec/pulls).

**Comment by msporny** [comment]
Migrated to https://github.com/w3c/did-core/pull/91. Closing.

**Pull request closed without merging**